discovery/spec.md

```diff
@@ -850,51 +846,59 @@ Content-Type: application/json
 
 [
   {
-    id: "{id}",
-    epoch: "{epoch}",
+    "id": "{id}",
+    "epoch": "{epoch}",
+    "name": "{name},
     ... remainder of Service attributes ...
   },
   ...
 ]
 ```
+
 #### `DELETE /services`
 
 This MUST delete all of the Services in the Discovery Endpoint that are
 contained in the request. If any of those Services cannot be deleted for any
-reason (e.g. an `id` is not given or a Service with the given `id` does not
-exist) then the entire request MUST fail with no effect.
+reason then the entire request MUST fail with no effect.
 
 If an `id` is not specified in any of the Services the entire request MUST
 fail. If no Service can be found that corresponds to a specified `id`, the
-entire request MUST fail. An `epoch` value MAY be omitted. If a Service
-corresponding to a specified `id` has a higher recorded `epoch` value than the
-one given the entire request MUST fail.
+processing MUST ignore that Service and not fail as a result of the Service
+already being deleted. An `epoch` value MAY be omitted in the incoming Service.
+If present, and is not greater than the Service's current `epoch` value, then
+the request MUST fail.
 
 If other service attributes are included in the request, those SHOULD be
 ignored for the purposes of processing the request.
 
-Service deletions MUST be processed in the order in which they are given. Thus,
-if a Service is specified multiple times the entire request MUST fail.
+There is no requirement that the Services in the request are processed in
+the order in which they appear.
+
+Upon successfully processing the request, the HTTP Response Body MUST include
+an array of Services that match the order of the array of Services specified
+in the request. The Services in the response MUST include at least the `id`
+attribute and SHOULD include the remaining Service attributes as they existed
+immediately prior to the Service being deleted, if possible. If the Service
+was deleted prior the processing of this request the Service values might not
+be available to be returned.
+
+If the response Services include an `epoch` value then each MUST be greater
+than the Service's value prior to the delete operation.
```

Comment:
I struggled with this because in the PUT/POST cases the returned Service contains the new epoch value, so from that perspective it seemed like this one should too. But it did feel weird to write that sentence :-)
Let's go with your semantics and see who complains. I don't like my new working... see what you think

Reply:
I have no objection to the language I just read and wouldn't block it.  However, just above that is ``The Services in the response MUST include at least the `id` attribute and SHOULD include the remaining Service attributes as they existed immediately prior to the Service being deleted, if possible. `` which I think already creates clarity.  I would recommend removal.